### ISSUE TITLE
Opensource Grafana Oncall mobile app

### ISSUE THREAD

**Issue by semekh** [grafana/oncall#3044]
### What would you like to see!

This is specifically interesting to us because we'd like to have a separate push notification channel that doesn't go through Grafana Oncall Cloud.

### Product Area

Mobile App

**Comment by imtoori** [comment]
Hi! What's the desired feature here?

You can already use the app with a self-hosted version of grafana oncall, you just need to connect to cloud for push notifications

Are you asking if there's a way of using the app without connecting to oncall cloud? 
In that case this is not possible at the moment and I don't really know if this is doable at all since the firebase configuration are hardcoded in the app and I don't think there's a way of configuring that dynamically. 
But I'll do some research in the next weeks if this is what's being requested

**Comment by semekh** [comment]
Yes, I'd like the self-hosted oncall to be able to push notifications without connecting to oncall cloud.
Since firebase configuration is hardcoded in the app, I would like to be able to compile the Android/iOS app with the customized firebase configuration. As an example, RocketChat does so and you can receive push notifications on your compiled app.

**Comment by schewara** [comment]
Hi

Just stumbled on this one, and it sounds like, that a solution like the one used in the Element Android Client would be a possible fit.

- https://github.com/vector-im/element-android/blob/develop/docs/unifiedpush.md

Unfortunately [UnifiedPush won't work on iOS](https://unifiedpush.org/users/faq/#will-unifiedpush-ever-work-on-ios)

**Comment by bmalynovytch** [comment]
> You can already use the app with a self-hosted version of grafana oncall, you just need to connect to cloud for push notifications

As far as I understand, having push notifications for a self-hosted Grafana Oncall instance requires (payable) seats on Grafana Cloud.
What's the purpose of having an OSS on-call management system, that can't notify people ? (or that requires paying the same expensive plans than full featured SaaS solution)

**Comment by cw-sanikachavan** [comment]
@Matvey-Kuk I am new to setting up grafana oncall, can you help me how to connect to grafana on call cloud for push notifications only and not for sms/phone

**Comment by Matvey-Kuk** [comment]
@cw-sanikachavan forwarding this question to @joeyorlando ;)

**Comment by joeyorlando** [comment]
hi there @cw-sanikachavan 👋 see the docs [here](https://grafana.com/docs/oncall/latest/set-up/open-source/#mobile-application-set-up) on how to setup the mobile app ✌️

**Comment by VermiumSifell** [comment]
> @bmalynovytch payable seats are required for SMS/Phone calls. For push notifications you could have as many OSS OnCall users as you want :)

Would still be quite good to be able to self-host it completely without connecting it to the cloud. Something like this maybe? https://unifiedpush.org/

**Comment by genofire** [comment]
- Unified-Push exists as issue #3181 
- To publish on F-Droid #3180 

i believe that is everything

**Comment by EStork09** [comment]
Is this something that Grafana is even willing to do or should this not be considered?
